.github/workflows/claude-assistant.yml

```diff
@@ -5,42 +5,122 @@ on:
+          the GitHub issue details and corresponding GitHub issue comment history (non-inclusive of the
+          latest, calling comment) below.
+          <github-issue-details>
+          ${{ steps.get-issue-details.outputs.issue-details }}
+          </github-issue-details>
+          <github-issue-comments>
+          ${{ steps.get-issue-details.outputs.issue-comments }}
+          </github-issue-comments>
+          <calling-comment>
+          ${{ github.event.comment.body }}
+          </calling-comment>
+      run: |
+        EOF=$(openssl rand -hex 16)
+        echo "message-content<<$EOF" >> $GITHUB_OUTPUT
+        echo "$MESSAGE_CONTENT" >> $GITHUB_OUTPUT
+        echo "$EOF" >> $GITHUB_OUTPUT
 
-    - name: Get Claude's response to issue comment
+    - name: Get Claude's response
       id: claude-response
-      if: steps.get-comment-body.outputs.comment != ''
       env:
+        MESSAGE_CONTENT: "${{ steps.generate-message.outputs.message-content }}"
         ANTHROPIC_API_KEY: ${{ secrets.ANTHROPIC_API_KEY }}
       run: |
-        echo response=$(curl https://api.anthropic.com/v1/messages \
-            --header "x-api-key: $ANTHROPIC_API_KEY" \
-            --header "anthropic-version: 2023-06-01" \
-            --header "content-type: application/json" \
-            --data \
-            '{
-                "model": "claude-3-opus-20240229",
-                "max_tokens": 1024,
-                "messages": [
-                    {"role": "user", "content": "${{ steps.get-comment-body.outputs.comment }}"}
-                ]
-            }') >> $GITHUB_OUTPUT
-
-    - name: Create comment
-      if: steps.get-comment-body.outputs.comment != '' && steps.claude-response.outcome == 'success'
+        MESSAGES_JSON=$(
+          jq -n \
+            --arg role "user" \
+            --arg content "$MESSAGE_CONTENT" \
+            '$ARGS.named'
+        )
+        MAX_TOKENS=1024
+        DATA_JSON=$(
+            jq -nr \
+              --arg model "claude-3-opus-20240229" \
+              --argjson max_tokens "$MAX_TOKENS" \
+              --argjson messages "[$MESSAGES_JSON]" \
+              '$ARGS.named'
+        )
+
+        echo response=$(
+          curl https://api.anthropic.com/v1/messages \
+               -H "x-api-key: $ANTHROPIC_API_KEY" \
+               -H "anthropic-version: 2023-06-01" \
+               -H "content-type: application/json" \
+               -d "$DATA_JSON"
+        ) >> $GITHUB_OUTPUT
+
+    - name: Create issue comment with Claude's error
+      uses: peter-evans/create-or-update-comment@v4
+      if: ${{ fromJson(steps.claude-response.outputs.response).error != '' }}
+      with:
+          issue-number: ${{ github.event.issue.number }}
+          body: |
+            **AI Assistant Claude**
+
+            ERROR
+
+            *Prompt sent to Claude:*
+            <details>
+            ${{ steps.generate-message.outputs.message-content }}
+            </details>
```

Comment:
I realise we may be in a weird scenario where every new prompt to Claude results in an entire record of the conversation being included in each of Claude's responses. Might be worth filtering out the "details" tag when sending follow up comments to Claude